lighthouse-viewer/app/src/drag-and-drop.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * Manages drag and drop file input for the page.
+ */
+class DragAndDrop {
+  /**
+   * @param {function(!File)} fileHandlerCallback Invoked when the user chooses a new file.
+   */
+  constructor(fileHandlerCallback) {
+    this._dropZone = document.querySelector('.drop_zone');
+    this._fileHandlerCallback = fileHandlerCallback;
+    this._dragging = false;
+
+    this._addListeners();
+  }
+
+  _addListeners() {
+    // The mouseleave event is more reliable than dragleave when the user drops
+    // the file outside the window.
+    document.addEventListener('mouseleave', _ => {
+      if (!this._dragging) {
+        return;
+      }
+      this._resetDraggingUI();
+    });
+
+    document.addEventListener('dragover', e => {
+      e.stopPropagation();
+      e.preventDefault();
```

Comment:
as an aside, what do each of these e.stopProgagation/preventDefaults do/why are they necessary? I built another drag/drop recently and I got away without some of them, but I don't know what exactly is required to pull this off...

cc: @ebidel for the original impl :)